### PR TITLE
Add error summary component

### PR DESCRIPTION
**Pull request by ethanWallace**
# Summary | Résumé

Add `gcds-error-summary` component to component library.

## Use

```html
<gcds-error-summary
  listen="true"        // boolean
  error-links=""      // string/object
  heading=""          // string
  sub-heading=""  // string
>
</gcds-error-summary>
```

### Listen

Using the `listen` attribute will listen for the `gcdsError` event and generate/display the error summary list when the parent form submits.

### Error-links

Use the `error-links` to render the error list without using event listeners. Pass a string or object formatted like the example below to render the list.
```
{
  "href": "Error message",
  "href": "Error message"
}
```

### Heading

To customize the error summary heading, pass a string to the `heading` attribute.

### Sub-heading

To customize the sub-heading, pass a string to the `sub-heading` attribute.

### CODE REVIEW

File: packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      }, 100);
+    }
+  }
+
+  /*
+  * Sort error object based on the order form compoennts appear in the form
+  */
+  sortErrors() {
+    let sortable = [];
+      for (var id in this.errorLinksObject) {
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx:118]: Can we avoid using `var` and use `let` or `const` instead?

File: packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          tabindex="-1"
+          ref={element => this.shadowElement = element as HTMLElement}
+          class={`gcds-error-summary ${Object.keys(errorQueue).length > 0 ? 'gcds-show' : ''}`}
+        >
+          <h2 class="summary__heading">
+            {heading ?
+              heading
+            :
+              i18n[lang].heading
+            }
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx:176]: Could we replace this with `{heading ?? i18n[lang].heading}`?

File: packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          </h2>
+          <p class="summary__sub-heading">
+            {subHeading ?
+              subHeading
+            :
+              i18n[lang].subheading
+            }
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx:183]: Could we replace this with `{subHeading ?? i18n[lang].subheading}`?

**Review by melaniebmn — CHANGES_REQUESTED** [review]
Can the error summary links scroll to the label for each field instead of the input field itself? I think that would be a slightly nicer interaction.

Aside from that, a couple of small comments but overall it looks great, nice work!

**Comment by ethanWallace** [comment]
> Can the error summary links scroll to the label for each field instead of the input field itself? I think that would be a slightly nicer interaction.

Good idea! I will look at adding this!

**Review by melaniebmn — APPROVED** [review]
LGTM!

**Pull request merged**